### PR TITLE
Reduce Case embeds + minor UI edits

### PR DESCRIPTION
**Pull request by alexkb0009**
So far the UI edits have been to migrate away from `Case.family` and `Case.secondary_families` to `Case.sample_processing.families` -
- `Case.family` will be kept, but only UUID kept, in order to select `canonicalFamily` (with all needed embeds) from `Case.sample_processing.families`.
- `Case.secondary_families` will be removed completely (**needs confirmation**). In place of it, memoized call of `Case.sample_processing.families.filter(function(family){ return family !== canonicalFamily; })` is used.

### Important Remaining to-dos
- Review the UI changes so far and check that any permissions we expect are handled
  - i.e. Can a Case exist with 0 families? Or can a Case exist with 1 or more families which might not be visible to all users who also have ability to see the Case?
- Figure out and move embeds around in Case.embedded_list on backend (if load this branch up now it should work but look broken / missing data).
- Find/test a Case with multiple families.
- Change the base of this branch to 'master' instead of `touchups-ui-2022-01c`, once the latter is merged.

### CODE REVIEW

File: src/encoded/types/case.py [hunk 1]
```diff
@@ -214,38 +235,10 @@ def _build_case_embedded_list():
...
-
-        # Individual linkTo
-        "sample_processing.families.proband.accession",
-        "sample_processing.families.proband.individual_id",
 
-        # Individual linkTo
-        "sample_processing.families.mother.accession",
-        "sample_processing.families.mother.individual_id",
+        # Sample Processing - Family[] linkTo
+        *_build_family_embeds(base_path='sample_processing.families'),
```
Comment (alexkb0009) [inline src/encoded/types/case.py:240]: Leading "*" is python spread operator, used it for readability (not have to split up list into multiple ones).

Comment (alexkb0009) [reply src/encoded/types/case.py:240]: Things which were defined below for "sample_processing.families" but not present in `_build_family_embeds` were added into _build_family_embeds. Might be able to trim _build_family_embeds down a tiny bit later.

**Review by Bianca-Morris — CHANGES_REQUESTED** [review]
After a lot of testing, I don't think this is ready for a merge. There are quite a few bugs with the pedigree visualization and cases. Here's a breakdown of what I found and what cases they were associated with:

- Broken pedigree lines for certain cases:
  - http://fourfront-cgaptest.9wzadzju3p.us-east-1.elasticbeanstalk.com/cases/GAPCAN3NPFYZ/
  - [Image from Gyazo](https://gyazo.com/4bab6291eb570c7cf386adeb8d0f62bc)
- Load errors on other cases, like this one:
  - http://fourfront-cgaptest.9wzadzju3p.us-east-1.elasticbeanstalk.com/cases/GAPCA1A5BQB3/
  - [Image from Gyazo](https://gyazo.com/1ab3f6c68171ff13af7dc26573405a2b)
- And this one:
  - http://fourfront-cgaptest.9wzadzju3p.us-east-1.elasticbeanstalk.com/cases/GAPCA1A5BQB3/
  - [Image from Gyazo](https://gyazo.com/2234ea85f50cc7aabed04f2d7b6d88d3)

**Comment by alexkb0009** [comment]
Went through cgap-devtest and seems OK now. But haven't been able to replicate the FileDownloadButton exception yet...

**Review by willronchetti — APPROVED** [review]
Looks great, seems to further improve case search performance on devtest

**Review by Bianca-Morris — APPROVED** [review]
Looks like the issues I ran into last time have been resolved. Seems good to go!

**Pull request merged**